var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -437,9 +450,9 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
 
         # set up MKLROOT for everyone using MKL package
         if '+mkl' in self.spec:
-            mkl_root = self.prefix.compilers_and_libraries.linux.mkl.lib.intel64  # noqa
+            mkl_root = self.intel_prefix.mkl.lib.intel64  # noqa
 
-            spack_env.set('MKLROOT', self.prefix)
+            spack_env.set('MKLROOT', mkl_root)
```

Comment:
@adamjstewart so when going through this...I kind of assumed this was a mistake.  Please advise